lua/luminate/init.lua

```diff
@@ -9,11 +9,13 @@ local M = {}
 local function set_highlight_groups()
   local highlight_groups = { 'yank', 'paste', 'undo', 'redo' }
   for _, group in ipairs(highlight_groups) do
-    highlight.set_highlight(config_module.config[group].hlgroup, {
-      ctermbg = config_module.config[group].ctermbg,
-      bg = config_module.config[group].guibg,
-      fg = config_module.config[group].fg,
-    })
+    if vim.api.nvim_get_hl(0, { name = config_module.config[group].hlgroup }) == nil then
```

Comment:
The vim.api.nvim_get_hl function does not return a nil value, so the subsequent code doesn't execute and set the highlight group.

To address this, I came up with the following code. What do you think?

local function is_empty_dict(tbl)
  return next(tbl) == nil
end

local function set_highlight_groups()
  local highlight_groups = { 'yank', 'paste', 'undo', 'redo' }
  for _, group in ipairs(highlight_groups) do
    local hlgroup = config_module.config[group].hlgroup
    local current_hl = vim.api.nvim_get_hl(0, { name = hlgroup })

    if is_empty_dict(current_hl) then
      highlight.set_highlight(hlgroup, {
        ctermbg = config_module.config[group].ctermbg,
        bg = config_module.config[group].guibg,
        fg = config_module.config[group].fg,
      })
    end
  end
end